plutus-streaming/src/Index/TxIdStatus.hs

```diff
@@ -0,0 +1,92 @@
+import Plutus.ChainIndex.Tx (ChainIndexTx (_citxTxId))
+import Plutus.ChainIndex.Types (BlockNumber (BlockNumber),
+                                TxConfirmedState (TxConfirmedState, blockAdded, timesConfirmed, validity),
+                                TxValidity (TxValid))
+import Plutus.Contract.CardanoAPI (fromCardanoTx)
+import Plutus.Streaming (ChainSyncEvent (RollForward), SimpleChainSyncEvent)
+
+type TxStatusIndex = SqliteIndex SimpleChainSyncEvent () TxId (Maybe TxConfirmedState)
+
+openIx :: FilePath -> IO TxStatusIndex
+openIx path =
+  fromJust <$> Ix.newBoxed query store onInsert 2000 2000 path
+
+-- Ignore notifications for now
+onInsert :: TxStatusIndex -> SimpleChainSyncEvent -> IO [()]
+onInsert _ _ = pure []
+
+-- No one will query this for now.
+query :: TxStatusIndex -> TxId -> [SimpleChainSyncEvent] -> IO (Maybe TxConfirmedState)
+query _ _ _ = pure Nothing
+
+store :: TxStatusIndex -> IO ()
+store ix = do
+  -- bufferedEvents <- getBuffer (ix ^. storage)
+  let c           = ix ^. handle
+      -- bufferedTxs = foldTxs $ getTxs . getBlocks <$> bufferedEvents
+  execute_ c "CREATE TABLE IF NOT EXISTS tx_state (txid TEXT PRIMARY KEY, confirmations INTEGER)"
+  -- execute_ handle "BEGIN"
+  -- forM_ (Map.assocs bufferedTxs) $
+  --   \(txid, v) -> execute handle "INSERT INTO tx_state (txid, confirmations) VALUES (?, ?)" (show txid, getSum $ timesConfirmed v)
+  -- This will really work your SSD to death, and it is not very useful, since
+  -- all txs that are persisted are settled.
+  -- execute handle "UPDATE tx_state SET confirmations = confirmations + ?" (Only $ Map.size bufferedTxs)
+  -- execute_ handle "COMMIT"
+
+getBlocks :: SimpleChainSyncEvent -> BlockInMode CardanoMode
+getBlocks (RollForward block _tip) = block
+getBlocks _                        = error "This should never happen"
+
+-- We won't have any rollbacks in the buffered events since those
+-- blocks have settled
+getTxs :: BlockInMode CardanoMode -> (BlockNo, [TxId])
+getTxs (BlockInMode (Block header transactions) era) =
+  case era of
+    C.ByronEraInCardanoMode   -> go header transactions era
+    C.ShelleyEraInCardanoMode -> go header transactions era
+    C.AllegraEraInCardanoMode -> go header transactions era
+    C.MaryEraInCardanoMode    -> go header transactions era
+    C.AlonzoEraInCardanoMode  -> go header transactions era
+  where
+    go :: forall era. C.IsCardanoEra era
+       => C.BlockHeader
+       -> [C.Tx era]
+       -> C.EraInMode era C.CardanoMode
+       -> (BlockNo, [TxId])
+    go (BlockHeader _ _ blockNo) txs era' =
+      (blockNo, _citxTxId <$> catMaybes (either (const Nothing) Just . fromCardanoTx era' <$> txs))
+
+foldTxs :: [(BlockNo, [TxId])] -> Map TxId TxConfirmedState
+foldTxs bs = snd $ foldl go (0, Map.empty) bs
+  where
+    go :: (Int, Map TxId TxConfirmedState)
+       -> (BlockNo, [TxId])
+       -> (Int, Map TxId TxConfirmedState)
+    go (confirmations, acc) (_, []) = (confirmations, acc)
```

Comment:
Do we need to be careful about strictness here? Perhaps better to have a strict datatype 

```
data GoArgs = GoArgs !Int !(Map TxId TxConfirmedState)`
```
so that go becomes `go :: GoArgs -> (BlockNo, [TxId]) -> GoArgs` and pattern match with `go !args ...` ?

This might not be a problem but I'd like to be sure we won't have a space leak here.